fix(common): inconsistency on TitleCasePipe

ottonascarella opened the pull request:
TitleCasePipe was not behaving as expected with on strings containing accents like é ó ú ã, etc.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->
TitleCasePipe was not working on strings containing accent characters. Ex: é ó å ä ü, etc, thanks to the way regex treats Unicode chars.

Closes to  #20123, #16586 (added its tests to make sure)

## What is the new behavior?
It works with those.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

Bug was found on https://br.udacity.com

trotyl commented:
Relates to #20123

ottonascarella commented:
yeah. it should close his case too, although his case should not close mine.
I will write his tests on mine too later on.

vicb commented on packages/common/src/pipes/case_conversion_pipes.ts:
```diff
@@ -23,19 +23,26 @@ export class LowerCasePipe implements PipeTransform {
       throw invalidPipeArgumentError(LowerCasePipe, value);
     }
-    return value.toLowerCase();
+    return value.toLocaleLowerCase();
```
If we use the `toLocaleLowerCase()` variant, we should inject the locale and use it.

ottonascarella replied:
I don't think it should be necessary. I can revert to using `toLowerCase` and `toUpperCase` which were working (regex was culprid)

vicb replied:
yep please revert then

vicb reviewed (changes requested):
Great change, thanks.

Please use the system wide locale.

gkalpak commented on packages/common/src/pipes/case_conversion_pipes.ts:
```diff
@@ -23,19 +23,26 @@ export class LowerCasePipe implements PipeTransform {
-  return word[0].toUpperCase() + word.substr(1).toLowerCase();
+function titleCase(str: string) {
+  if (!str) return str;
+  return str.split(/\s/g)
```
Just wanted to point out that this is potentially a BC. E.g. `reg,green,blue` would previously become `Reg,Green,Blue`.
Since apparently the rules for title-casing aren't standardized, it might be a good idea to be explicit about what this pipe does.

gkalpak commented on packages/common/src/pipes/case_conversion_pipes.ts:
```diff
@@ -23,19 +23,26 @@ export class LowerCasePipe implements PipeTransform {
+        }
+        return word;
+      })
+      .join(' ');
```
This will replace all whitespace characters (e.g. `\t` or `\n`) with ` `, which is unexpected (even if uncommon :smiley:).

vicb commented:
This is a change in behavior (wrt to the aaa,bbb,ccc mentioned by @gkalpak) and the implementation could be simpler.
I think we need to discuss that with the team.

Commit:
fix(common):  inconsistency on TitleCasePipe

TitleCasePipe was not behaving as expected with on strings containing accents like é ó ú ã, etc.

refactor(common): TitleCasePipe not using localeLowerCase/localeUpperCase

MattiJarvinen-BA commented:
I don't even get why use title case pipe. 
CSS has `text-transform: capitalize;`
https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform

gkalpak commented:
Handling accent characters has been fixed by #22600.
AFAICT, this is obsolete now.

##
WRT https://github.com/angular/angular/pull/22322#issuecomment-368772636, it would be fun to make a titlecase pipe that is compatible with the CSS spec :grin:
